### DISCUSSION TITLE
What is going on with `type(somevar)` ?

### DISCUSSION THREAD

**Discussion by johann-petrak** [exaloop/codon#273] [General]
Example: 

```
print(type(123))
# prints: (item1: (), item2: ())
```

Hard to find information about this in the docs, and in general the docs do not seem to go into much detail about differences between python and codon or at least the details are not easy to find. 

For example, there is a hint that codon supports only ASCII strings and not Python unicode codepoints, which, I think is a real problem and I wanted to read more about this but could not find it. 
Coming from an NLP background I think not supporting unicode properly (it is 2023!)  would be a major problem for any language, really.

**Comment by elisbyberi** [comment]
@johann-petrak `# prints: (item1: (), item2: ())` is duplicate of #170

**Comment by inumanag** [comment]
We do not support first-class types; `type(var)` is just a shortcut to get a type variable where it is needed (e.g. `List[type(a)]`). This will be revamped in the following releases.

No, we do not yet support Unicode. This is something that we plan to address as well and are looking for contributions. FWIW Codon originated as a high-performance genomics sequencing DSL where Unicode was not really needed (actually, it was detrimental to performance), and it stuck ever since.
